### PR TITLE
DAOS-3627 md: pool destroy: resume destroy/connected testing

### PR DESCRIPTION
**Pull request by kccain**
Pool destroy tests test_destroy_connected and test_destroy_withdata
are re-introduced with this change. The tests had previously been
skipped in the testing flow due to bugs with pool destroy that have
now been been resolved in a prior commit for DAOS-3627.

Quick-build: true
Skip-run_test: true
Test-tag: pr,-hw,pool,destroy
Skip-func-hw-test: true

Signed-off-by: Ken Cain <kenneth.c.cain@intel.com>

### CODE REVIEW

**Comment by kccain** [comment]
This commit will occur after landing the associated DAOS implementation change PR 1625. The branch that this PR is currently based off is kccain/daos_3627 that contains that implementation change. The testing results demonstrate that the 2 re-enabled tests will pass functional testing with the DAOS change in place.

**Comment by kccain** [comment]
Changed base to master now that PR 1625 associated DAOS implementation change has landed. In the changed files this is the one specific to this PR:
src/tests/ftest/pool/destroy_tests.py

File: src/tests/ftest/pool/destroy_tests.py [hunk 1]
```diff
@@ -322,7 +322,6 @@ def test_destroy_wrong_group(self):
                 self.pool.uuid, group_names[0]),
             False)
 
-    @skipForTicket("DAOS-2741")
     def test_destroy_connected(self):
         """Destroy pool with connected client.
```
Comment (mjean308) [inline src/tests/ftest/pool/destroy_tests.py]: I updated to this PR and ran the test and verified that the pool does not get destroyed when connected.  When trying to destroy the connected pool DAOS reported the expected error as DER_BUSY.

Attempting to destroy a connected pool
Expected exception - destroying connected pool: Pool destroy returned non-zero. RC: -1012

Comment (ravalsam) [reply src/tests/ftest/pool/destroy_tests.py]: oh, In PR same  test is passing [https://build.hpdd.intel.com/job/daos-stack/job/daos/view/change-requests/job/PR-1670/3/artifact/Functional/job-2020-01-22T17.20-55d3c15-pool-destroy_tests/job.log](https://build.hpdd.intel.com/job/daos-stack/job/daos/view/change-requests/job/PR-1670/3/artifact/Functional/job-2020-01-22T17.20-55d3c15-pool-destroy_tests/job.log)

File: src/tests/ftest/pool/destroy_tests.py [hunk 2]
```diff
@@ -373,7 +372,6 @@ def test_destroy_connected(self):
         self.assertTrue(
             exception_detected, "No exception when deleting a connected pool")
 
-    @skipForTicket("DAOS-2741")
     def test_destroy_withdata(self):
```
Comment (mjean308) [inline src/tests/ftest/pool/destroy_tests.py]: The destroywithdata variant also reports the correct error when trying to destroy the pool.  DER_BUSY

Opening container 63369698-9EF9-4E54-B8EE-5A92C9A137CA
[stdout] DEBUG 19:47:44.344564 drpc_server.go:63: Closing a session
Writing 1 object(s), with 1 record(s) of 9 bytes(s) each, in container 63369698-9EF9-4E54-B8EE-5A92C9A137CA with object class OC_S1
Attempting to destroy a connected pool with data
Expected exception - destroying connected pool with data: Pool destroy returned non-zero. RC: -1012

**Comment by kccain** [comment]
@daos-stack/daos-gatekeeper this PR was ready to land a while ago but I've forgotten about it in the meantime. Could someone take a look at it?

**Comment by kccain** [comment]
Merged master after discussion with @jolivier23 . Re-ran and again passed the applicable destroy_tests.py functional tests. @daos-stack/daos-gatekeeper this is ready to land.

**Pull request merged**